users/serializers.py

```diff
@@ -11,3 +13,31 @@ class CurrentUserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
         fields = ('username', )
+
+
+class SocialTokenObtainPairSerializer(serializers.Serializer):
+    """Serializes the token data for social user. """
+    def __init__(self, *args, **kwargs):
+        super(SocialTokenObtainPairSerializer, self).__init__(*args, **kwargs)
+
+        self.fields['username'] = serializers.CharField()
+
+    def validate(self, attrs):
+        data = super(SocialTokenObtainPairSerializer, self).validate(attrs)
+        try:
+            self.user = User.objects.get(username=attrs['username'])
+        except User.DoesNotExist:
```

Comment:
According to the pipelines, catching the exception is redundant since the user will be created if doesn't exist.